Intermittent error from the new Harvesting Clients test

Has been observed a couple of times on Jenkins since the PR #9227 was merged. Meaning, it's intermittent, the test passes on most runs. 
The error happens in the new test that's somewhat extensive and runs a real harvest of an external archive. However, the 500 below happens before anything heavy is attempted, merely when creating the new harvesting client: 
```
java.lang.AssertionError: expected:201 but was:500
at edu.harvard.iq.dataverse.api.HarvestingClientsIT.testHarvestingClientRun(HarvestingClients0T.java:184)
```
which is really a cheapest operation you can think of. This makes me think that this is another weird timing condition caused by an overall load of a full restassured run on the system. I expect it to be straightforward to fix/work around.

We had a related failure on develop yesterday: https://jenkins.dataverse.org/job/IQSS-dataverse-develop/1306/consoleFull

"[ERROR] Failures: ", "[ERROR]   HarvestingClientsIT.testCreateEditDeleteClient:109 expected:<201> but was:<500>",

For the line number, develop is e12f3e6

--- a/src/test/java/edu/harvard/iq/dataverse/api/HarvestingClientsIT.java
+++ b/src/test/java/edu/harvard/iq/dataverse/api/HarvestingClientsIT.java
@@ -79,7 +79,7 @@ public class HarvestingClientsIT {
         // This method focuses on testing the native Dataverse harvesting client
         // API. 
         
-        String nickName = UtilIT.getRandomString(6);
+        String nickName = "h" + UtilIT.getRandomString(6);
         
 
         String clientApiPath = String.format(HARVEST_CLIENTS_API+"%s", nickName);
@@ -100,7 +100,6 @@ public class HarvestingClientsIT {
         assertEquals(UNAUTHORIZED.getStatusCode(), rCreate.getStatusCode());
 
         
-        Thread.sleep(1000L);
         // Try to create the same as admin user, should succeed:
         
         rCreate = given()
@@ -167,7 +166,7 @@ public class HarvestingClientsIT {
         // method, we don't need to pay too much attention to this method, aside 
         // from confirming the expected HTTP status code.
         
-        String nickName = UtilIT.getRandomString(6);
+        String nickName = "h" + UtilIT.getRandomString(6);
 
         String clientApiPath = String.format(HARVEST_CLIENTS_API+"%s", nickName);
         String clientJson = String.format("{\"dataverseAlias\":\"%s\","
@@ -178,7 +177,6 @@ public class HarvestingClientsIT {
                 + "\"metadataFormat\":\"%s\"}", 
                 harvestCollectionAlias, HARVEST_URL, ARCHIVE_URL, CONTROL_OAI_SET, HARVEST_METADATA_FORMAT);
         
-        Thread.sleep(1000L);
         Response createResponse = given()
                 .header(UtilIT.API_TOKEN_HTTP_HEADER, adminUserAPIKey)
                 .body(clientJson)

